core/trino-main/src/main/java/io/trino/server/TrinoSystemRequirements.java

```diff
@@ -93,7 +93,7 @@ else if ("Mac OS X".equals(osName)) {
 
     private static void verifyJavaVersion()
     {
-        Version required = Version.parse("17.0.3");
+        Version required = Version.parse("17.0.4");
```

Comment:
The commit message doesn't match this change. `air.java.version` is unchanged, and there are actually three separate versions here, which we should be careful about not conflating:

* minimum required version for users to run Trino
* minimum required version for building Trino
* the version used for Docker images

Plus whatever else might use `.java-version` (I manage my Java versions manually so I'm not familiar with exactly how the version managers work)

I would be fine with making the last two the same, as it seems reasonable for developers to build and test against the version that we ship in the Docker image. But we need to be careful about bumping the required Java version, as that can impact users who manually install Java and might not need the latest version. We typically only bump it if there are crash bugs or other egregious issues. Though I'm open to discussing changing this policy to be more aggressive if it avoids support issues than it causes.

Reply:
Just to clarify, when you say "minimum required version for building Trino", you're referring to the version of the JDK used to compile Trino, not the version of the JDK the code relies on (i.e., language level, Java APIs, JVM features), right?

Reply:
@martint we have 4 different source of the JDK version:

1) `air.java.version` - maven check whether minimum JDK version is used to compile code
2) `TrinoSystemRequirements` - runtime check whether minimum JDK verion is used to compile code
3) `project.build.targetJdk` - language level (`--release` passed to the compiler)
4) `.java-version` - JDK version to used for local development/docker image